storage/files.rb

```diff
@@ -382,6 +386,78 @@ def generate_signed_url project_id:, bucket_name:, file_name:
   # [END generate_signed_url]
 end
 
+def set_event_based_hold project_id:, bucket_name:, file_name:
+  # [START storage_set_event_based_hold]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Cloud Storage bucket"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+  file    = bucket.file file_name
+
+  file.set_event_based_hold!
+
+  puts "Event based hold was set for #{file_name}"
+  # [END storage_set_event_based_hold]
+end
+
+def release_event_based_hold project_id:, bucket_name:, file_name:
+  # [START storage_release_event_based_hold]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Cloud Storage bucket"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+  file    = bucket.file file_name
+
+  file.release_event_based_hold!
+
+  puts "Event based hold was released for #{file_name}"
+  # [END storage_release_event_based_hold]
+end
+
+def set_temporary_hold project_id:, bucket_name:, file_name:
+  # [START storage_set_temporary_hold]
+  # project_id  = "Your Google Cloud project ID"
+  # bucket_name = "Your Google Cloud Storage bucket name"
+  # file_name   = "Name of a file in the Cloud Storage bucket"
+
+  require "google/cloud/storage"
+
+  storage = Google::Cloud::Storage.new project: project_id
+  bucket  = storage.bucket bucket_name
+  file    = bucket.file file_name
+
+  file.set_temporary_hold!
+
+  puts "Temporary hold was set for #{file_name}"
```

Comment:
This sentence is missing a period at the end.

Reply:
Wanted to keep it contextual to the sample.